internal/overlord/servstate/manager.go

```diff
@@ -338,20 +352,28 @@ func (m *ServiceManager) Replan() ([]string, []string, error) {
 	}
 	defer releasePlan()
 
+	m.servicesLock.Lock()
+	defer m.servicesLock.Unlock()
+
 	needsRestart := make(map[string]bool)
 	var stop []string
-	for name, activeService := range m.services {
-		if currentService, ok := m.plan.Services[name]; ok &&
-			currentService.Equal(activeService.originalPlan) {
-			continue
+	for name, s := range m.services {
+		s.lock.Lock()
```

Comment:
Related to earlier comment: what's the benefit of having individual locks on every service?  Wouldn't it be enough to have a more encompassing lock similar to what we had before (servicesLock), since these are just management control tasks and not fast-paced or necessarily concurrent ones?  Easier to get right if so.

Reply:
Per above, I've merged everything into one `servicesLock`.